### ISSUE TITLE
IndexError: list index out of range

### ISSUE THREAD

**Issue by Atisom** [ngoduykhanh/PowerDNS-Admin#833]
Hello,

PDNS server: 4.3.0-1pdns.buster
PowerDNS admin: ngoduykhanh/powerdns-admin:0.2.2  from docker hub

We got "500 Oops! Something went wrong" error message when we want to open some zone. In the log:

```
[2020-10-29 13:12:42 +0000] [11] [ERROR] Exception on /domain/example-domain.org [GET]
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/flask/app.py", line 2446, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/lib/python3.8/site-packages/flask/app.py", line 1951, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/lib/python3.8/site-packages/flask/app.py", line 1820, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/lib/python3.8/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/usr/lib/python3.8/site-packages/flask/app.py", line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/lib/python3.8/site-packages/flask/app.py", line 1935, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/usr/lib/python3.8/site-packages/flask_login/utils.py", line 261, in decorated_view
    return func(*args, **kwargs)
  File "/app/powerdnsadmin/decorators.py", line 60, in decorated_function
    return f(*args, **kwargs)
  File "/app/powerdnsadmin/routes/domain.py", line 108, in domain
    comment=r['comments'][index]['content']
IndexError: list index out of range
[2020-10-29 13:12:42,439] [app.py:1890] ERROR - Exception on /domain/example-domain.org [GET]
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/flask/app.py", line 2446, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/lib/python3.8/site-packages/flask/app.py", line 1951, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/lib/python3.8/site-packages/flask/app.py", line 1820, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/lib/python3.8/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/usr/lib/python3.8/site-packages/flask/app.py", line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/lib/python3.8/site-packages/flask/app.py", line 1935, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/usr/lib/python3.8/site-packages/flask_login/utils.py", line 261, in decorated_view
    return func(*args, **kwargs)
  File "/app/powerdnsadmin/decorators.py", line 60, in decorated_function
    return f(*args, **kwargs)
  File "/app/powerdnsadmin/routes/domain.py", line 108, in domain
    comment=r['comments'][index]['content']
IndexError: list index out of range
```

When I get this zone by curl:
```
curl -H "X-API-Key: TOPsecret" https://dns-master.org/api/v1/servers/localhost/zones/example-domain.org
{
	"account": "",
	"api_rectify": false,
	"dnssec": false,
	"edited_serial": 2020102902,
	"id": "example-domain.org.",
	"kind": "Native",
	"last_check": 0,
	"master_tsig_key_ids": [],
	"masters": [],
	"name": "example-domain.org.",
	"notified_serial": 0,
	"nsec3narrow": false,
	"nsec3param": "",
	"rrsets": [{
			"comments": [{
					"account": "",
					"content": "Ticket#123456",
					"modified_at": 1603276587
				}
			],
			"name": "example-domain.org.",
			"records": [{
					"content": "\"google-site-verification=xx123\"",
					"disabled": false
				}, {
					"content": "\"X=123456\"",
					"disabled": false
				}
			],
			"ttl": 3600,
			"type": "TXT"
		}, {
			"comments": [],
			"name": "example-domain.org.",
			"records": [{
					"content": "1 aspmx.l.google.com.",
					"disabled": false
				}, {
					"content": "5 alt1.aspmx.l.google.com.",
					"disabled": false
				}, {
					"content": "5 alt2.aspmx.l.google.com.",
					"disabled": false
				}, {
					"content": "10 alt3.aspmx.l.google.com.",
					"disabled": false
				}, {
					"content": "10 alt4.aspmx.l.google.com.",
					"disabled": false
				}
			],
			"ttl": 86400,
			"type": "MX"
		}, {
			"comments": [],
			"name": "example-domain.org.",
			"records": [{
					"content": "dns-master.org. hostmaster.master.org. 2020102901 10800 3600 604800 3600",
					"disabled": false
				}
			],
			"ttl": 86400,
			"type": "SOA"
		}, {
			"comments": [],
			"name": "example-domain.org.",
			"records": [{
					"content": "dns-master.org.",
					"disabled": false
				}, {
					"content": "dns-slave.org.",
					"disabled": false
				}
			],
			"ttl": 86400,
			"type": "NS"
		}, {
			"comments": [],
			"name": "example-domain.org.",
			"records": [{
					"content": "1.2.3.4",
					"disabled": false
				}
			],
			"ttl": 3600,
			"type": "A"
		}
	],
	"serial": 2020102901,
	"slave_tsig_key_ids": [],
	"soa_edit": "",
	"soa_edit_api": "DEFAULT",
	"url": "/api/v1/servers/localhost/zones/example-domain.org."
}
```

What do you think, what can cause this?

Thank you,
Attila

**Comment by christianbur** [comment]
I mean I had the same problem before. I could fix it by opening the database of powerdns directly and removing the comments.

**Comment by Atisom** [comment]
fixed in v0.2.3
